### PR TITLE
Support for custom protocol stream + test support

### PR DESCRIPTION
**Pull request by mikiquantum**
Fixes: https://github.com/centrifuge/go-centrifuge/issues/487

### CODE REVIEW

**Comment by mikiquantum** [comment]
@vimukthi-git Adding tests here requires to configure dep, travis and so on. Can we create another issue to tackle that?

**Comment by vimukthi-git** [comment]
thats only for CI right? just running a test locally we don't need that. 
Even then I think we can take time and do the right way since by not adding tests we are postponing the problem and making it harder to fix problems down the line.

**Comment by mikiquantum** [comment]
travis for CI yes, but dep needs to be configured as many libraries need to be available during test runs, disregarding if you run it in travis or locally. I could argue if we do this right we have travis testing the changes otherwise all this "locally works" becomes the source of truth 😄

**Comment by vimukthi-git** [comment]
I also don't want locally works philosophy, as I said right way is to add tests and also configure CI. 
If you are ok with it I can take this pr and make the necessary changes and add tests + ci :)

**Comment by lucasvo** [comment]
@mikiquantum, given we're not in a rush merging this ASAP, why don't we add CI & tests in a separate PR and then merge this PR once we have tests. WDYT?

**Comment by mikiquantum** [comment]
I am adding some tests now. We can create a PR later for CI.

**Comment by lucasvo** [comment]
@mikiquantum, given we're not in a rush merging this ASAP, why don't we add CI & tests in a separate PR and then merge this PR once we have tests. WDYT?

File: host.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	mrand "math/rand"
+
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// MakeBasicHost creates a LibP2P host with a random peer ID listening on the
+// given multiaddress. It will use secio if secio is true.
+func MakeBasicHost(listenPort int, secio bool, randseed int64) (host.Host, error) {
```
Comment (vedhavyas) [inline host.go]: Not sure if having this part of the library is a good idea. This function is assuming tcp as a default protocol

Comment (mikiquantum) [reply host.go]: It is assuming many things not only tcp but protocols, key types ..., only used in tests and in the examples folders.

**Pull request merged**